packages/local-environment/src/domain.ts

```diff
@@ -24,15 +25,29 @@ export class NomadDomain {
   network: Network;
   metricsPort?: number;
   nomadEnv?: NomadEnv;
+  forkUrl?: string;
+  forkBlockheight?: number;
+  docker: Dockerrode;
 
   connectedNetworks: NomadDomain[];
 
-  constructor(network: Network, nomadEnv?: NomadEnv) {
+  constructor(name: string, domain: number, forkUrl?: string, docker?: Dockerrode, nomadEnv?: NomadEnv) {
```

Comment:
I don't like that Docker is moved inside of NomadDomain. I think there should be another level of abstraction. In particular, forkUrl and block height is only used to create another network within the constructor and then it is stored into the domain for some reason.

If we need docker (and other stuff) in here u need to extend the Domain class with those traits. This will allow u keeping the original Domain being constructed from a network.

```
class DockerForkedDomain extends NomadDomain {
  constructor(name, domain, ....)
    const network = new HardhatOrForkedNetwork_ForInstance(...);
    super(network,...)
```

Reply:
As prescribed, DockerForkedDomain will now be a new abstraction completely SEPARATE from the default hardhat networks. Hardhat networks will now be purely for localhosted testing - alternative settings for docker, hardhat forking etc will go in this DockerForkedDomain abstraction.

Reply:
Note: DockerForkedDomain will extend NomadDomain.